include/RAJA/policy/cuda/reduce.hpp

```diff
@@ -70,21 +72,99 @@
 
 #include <cuda.h>
 
+#if defined(RAJA_ENABLE_OPENMP)
+#include <omp.h>
+#endif
+
 namespace RAJA
 {
 
-// HIDDEN namespace to encapsulate helper functions
-namespace HIDDEN
+namespace reduce
+{
+
+template <typename T>
+struct atomic_operators
+{
+
+};
+
+template <typename T>
+struct atomic_operators<sum<T>>
+{
+  RAJA_HOST_DEVICE RAJA_INLINE
+  void operator()(T &val, const T v)
+  {
+#ifdef __CUDA_ARCH__
+    RAJA::_atomicAdd(&val, v);
+#else
+    val += v;
```

Comment:
specializations do use pattern/reduce.hpp, I'll look into moving atomics there although they already share that namespace 

template <size_t BLOCK_SIZE, bool Async, typename T>
struct ReduceMaxLoc<cuda_reduce<BLOCK_SIZE, Async>, T>
    : public CudaReduceLoc<Async, RAJA::reduce::maxloc<T, Index_type>, T, Index_type> {